### PR TITLE
Extract error handling for hooks into a reusable form

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/RebaseOnParentBackgroundable.java [hunk 1]
```diff
@@ -99,41 +95,8 @@ public void doRun(ProgressIndicator indicator) {
       return;
     }
 
-    AtomicReference<Try<@Nullable ExecutionResult>> wrapper = new AtomicReference<>(
-        Try.success(null));
-
     val preRebaseHookExecutor = new PreRebaseHookExecutor(gitRepository);
-
-    new GitFreezingProcess(project, getTitle(), () -> {
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/RebaseOnParentBackgroundable.java:107]: Whoops let's keep `GitFreezingProcess`... when someone e.g. triggers a commit (Cmd+K), this is what happens when an action is wrapped into `GitFreezingProcess`:

image

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/RebaseOnParentBackgroundable.java:107]: ...which is desirable ofc

Comment (mkondratek) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/RebaseOnParentBackgroundable.java:107]: wow, thanks for testing & verifying

File: backend/api/src/main/java/com/virtuslab/gitmachete/backend/hooks/OnExecutionTimeout.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 package com.virtuslab.gitmachete.backend.hooks;
 
 public enum OnExecutionTimeout {
-  THROW, RETURN_NULL
+  RETURN_NULL, THROW_EXCEPTION
```
Comment (mkondratek) [inline backend/api/src/main/java/com/virtuslab/gitmachete/backend/hooks/OnExecutionTimeout.java:4]: iirc in c++ one is able to throw anything (e.g. `int`) but in java this specification seems to be redundant indeed

**Review by mkondratek — APPROVED** [review]
okidoki

**Pull request merged**